programs/keeper-client/Commands.cpp

```diff
@@ -475,4 +475,45 @@ void FourLetterWordCommand::execute(const ASTKeeperQuery * query, KeeperClient *
     std::cout << client->executeFourLetterCommand(query->args[0].safeGet<String>()) << "\n";
 }
 
+bool GetAllChildrenNumberCommand::parse(IParser::Pos & pos, std::shared_ptr<ASTKeeperQuery> & node, Expected & expected) const
+{
+    String path;
+    if (!parseKeeperPath(pos, expected, path))
+        path = ".";
+
+    node->args.push_back(std::move(path));
+
+    return true;
+}
+
+void GetAllChildrenNumberCommand::execute(const ASTKeeperQuery * query, KeeperClient * client) const
+{
+    auto path = client->getAbsolutePath(query->args[0].safeGet<String>());
+
+    std::queue<fs::path> queue;
+    queue.push(path);
+    Coordination::Stat stat;
+    client->zookeeper->get(path, &stat);
+
+    int totalNumChildren = stat.numChildren;
+    while (!queue.empty())
+    {
+        auto next_path = queue.front();
+        queue.pop();
+
+        auto children = client->zookeeper->getChildren(next_path);
+        std::transform(children.cbegin(), children.cend(), children.begin(), [&](const String & child) { return next_path / child; });
```

Comment:
Please use for-range loop because it's more readable.
I know it's the same but for loop is preferred in our codebase.

Reply:
void FindBigFamily::execute(const ASTKeeperQuery * query, KeeperClient * client) const
{
    auto path = client->getAbsolutePath(query->args[0].safeGet<String>());
    auto n = query->args[1].safeGet<UInt64>();

    std::vector<std::tuple<Int32, String>> result;

    std::queue<fs::path> queue;
    queue.push(path);
    while (!queue.empty())
    {
        auto next_path = queue.front();
        queue.pop();

        auto children = client->zookeeper->getChildren(next_path);
        **std::transform(children.cbegin(), children.cend(), children.begin(), [&](const String & child) { return next_path / child; });**

        auto response = client->zookeeper->get(children);

        for (size_t i = 0; i < response.size(); ++i)
        {
            result.emplace_back(response[i].stat.numChildren, children[i]);
            queue.push(children[i]);
        }
    }

    std::sort(result.begin(), result.end(), std::greater());
    for (UInt64 i = 0; i < std::min(result.size(), static_cast<size_t>(n)); ++i)
        std::cout << std::get<1>(result[i]) << "\t" << std::get<0>(result[i]) << "\n";
}



Do we also need to change this method and this place together?

Reply:
I would change it, and add a method for it

Reply:
std::transform(children.cbegin(), children.cend(), children.begin(), [&](const String & child) { return next_path / child; })
=>  
for (auto childrenNode = children.begin(); childrenNode != children.end(); ++childrenNode)
        {
            const String & child = *childrenNode;
            *childrenNode = next_path / child;
        }

What do you think? I tested OK.

Reply:
I meant
```
for (auto & child = children)
    child = next_path / child;
```

Reply:
okay. to for (auto & child : children)
            child = next_path / child;  


 I tested find_big_family and get_all_children_number OK.